File: Questions/Min Cost Climbing Stairs.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int size = cost.size();
        cost.push_back(0);

        for(int i=2; i <= size; i++){
            cost[i] += min(cost[i-1], cost[i-2]);
        }
        return cost[size];
    }
}; 

File: Leetcode Daily/Min COst Climbing Stairs.rs
impl Solution {
    pub fn min_cost_climbing_stairs(cost: Vec<i32>) -> i32 {
        let mut prev1=0;
        let mut prev2=0;

        for &i in cost.iter(){
            let c = i + prev1.min(prev2);
            prev2 = prev1;
            prev1 = c;
        }

        prev1.min(prev2)
    }
}
